src/System.Management.Automation/engine/PropertyAccessor.cs

```diff
@@ -166,11 +166,11 @@ internal JsonConfigFileAccessor()
         /// not interfere with PowerShell initialization
         /// </summary>
         /// <returns>Returns the directory if present or creatable. Throws otherwise.</returns>
-        private string GetAppDataConfigDirectory()
+        private string GetCurrentUserConfigDirectory()
         {
             if (null == appDataConfigDirectory)
```

Comment:
I think we should not create this directory until we have to. That means all read operations should not try creating this directory, and only write operations to the user configuration file would trigger creation of this directory.